### ISSUE TITLE
Why remote the content-type when requestBody is empty ?

### ISSUE THREAD

**Issue by zWingz** [axios/axios#5814]
### Describe the issue

I have set `headers: { 'content-type': 'application/json' }`. But It can't be send to the server.
And I found the code https://github.com/axios/axios/blob/v1.x/lib/adapters/xhr.js#L195 will remove the `content-type` when the request body is empty.

### Expected behavior

Still send the `content-type` to the server ?

### Axios Version

1.3.3

### Adapter Version

xhr

### Browser

chrome

### Browser Version

latest

**Comment by NikoRaisanen** [comment]
Hey zWingz, what's the use case for defining a content type if there is no content?

**Comment by zWingz** [comment]
<img width="981" alt="企业微信截图_adf76f5a-6867-4f84-a2d5-7a7dd721dd4e" src="https://github.com/axios/axios/assets/13031838/2515c965-08f3-406e-b15b-90aff59080f2">
@NikoRaisanen Spring Boot will check the `content-type`

**Comment by jasonsaayman** [comment]
may want to check the RFC https://www.rfc-editor.org/rfc/rfc7231#section-4.3.3 it is accepted that a post request would at least have a body of `{}` if it was nothing and not just no body at all

**Comment by prajwalbandak** [comment]
Hey @NikoRaisanen , would love to work on this issue. Can you please assign to me?

**Comment by tlevesque-ueat** [comment]
I have the opposite problem. When I try to send a POST with no body, the request is sent with a `Content-Type` of `application/x-www-form-urlencoded`. This value is never set explicitly in my code, I don't know where it comes from. And I can't find a way to completely suppress the header. Explicitly setting it to `undefined` has no effect. Setting it to `null` or an empty string sends an empty header. Is there a way to remove it?

**Comment by DigitalBrainJS** [comment]
@tlevesque-ueat 

 > Is there a way to remove it?

Just set it to `false`

**Comment by tlevesque-ueat** [comment]
> Just set Content-Type to false

As far as I can tell, it does the same thing as setting it to null or an empty string (sends an empty header).
Doesn't seem to be a problem in my scenario, though, so I'll stick to that. Thanks

**Comment by DigitalBrainJS** [comment]
@tlevesque-ueat The `false` value completely removes the header from the set and prevents Axios from setting it in the send sequence.

**Comment by tlevesque-ueat** [comment]
Yes, I realize that now. The tool I was using to inspect the request (webhook.site) was showing an empty Content-Type, but it doesn't seem to actually be present in the request. Thanks!

**Comment by haug1** [comment]
This was quite confusing when I stumbled upon it.
Yes, sure, it's not technically good practice to send the `content-type` header without a request body, but sometimes you rely on APIs that are not well made. When those situations occur, you don't want your http library to block your path forward.

This line of code is too opinionated and can actually completely block users from accomplishing what they want to do with `axios`, please remove:
https://github.com/axios/axios/blob/2a9763426e43d996fd60d01afe63fa6e1f5b4fca/lib/adapters/xhr.js#L137

**Comment by DigitalBrainJS** [comment]
> can actually completely block users from accomplishing what they want

As that line states, it removes the `content-type` header for an `undefined` body only, not an empty one, so there are no blockers to define an empty body to keep the header.

```js
axios.post("https://httpbin.org/post", "", {
    headers: {
      "Content-Type": 'application/octet-stream'
    }
}); // an empty string

axios.post('http://httpbin.org/post', new Uint8Array(0), {
    headers: {
      "Content-Type": 'application/json'
    }
}); // an empty buffer for the case if you wanna send 0 bytes body (not null / "" / {}) with 'application/json'

axios.post("https://httpbin.org/post", null, {
    headers: {
      "Content-Type": 'application/octet-stream'
    }
}); // not recommended, as object payload rendering depends on content-type value (json, formdata, urlencoded)
```

**Comment by haug1** [comment]
> > can actually completely block users from accomplishing what they want
> 
> As that line states, it removes the `content-type` header for an `undefined` body only, not an empty one, so there are no blockers to define an empty body to keep the header.
> 
> axios.post("https://httpbin.org/post", "", {
>     headers: {
>       "Content-Type": 'application/octet-stream'
>     }
> }); // an empty string
> 
> axios.post('http://httpbin.org/post', new Uint8Array(0), {
>     headers: {
>       "Content-Type": 'application/json'
>     }
> }); // an empty buffer for the case if you wanna send 0 bytes body (not null / "" / {}) with 'application/json'
> 
> axios.post("https://httpbin.org/post", null, {
>     headers: {
>       "Content-Type": 'application/octet-stream'
>     }
> }); // not recommended, as object payload rendering depends on content-type value (json, formdata, urlencoded)

I guess I didn't really give that a thought, it's a good point.
However, I still think it's weird that axios silently manipulates the request like this, but yeah, that is a feasible workaround after you understand what's happening.
I understand that the best approach might not be to remove this line of code because of backwards-compatibility.
